pkg/providers/v1/aws.go

```diff
@@ -4147,15 +4147,27 @@ func (c *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, apiS
 			loadBalancerName,
 			v2Mappings,
 			instanceIDs,
-			subnetIDs,
+			discoveredSubnetIDs,
 			internalELB,
 			annotations,
 		)
 		if err != nil {
 			return nil, err
 		}
 
-		subnetCidrs, err := c.getSubnetCidrs(subnetIDs)
+		// if the LB has subnet, use the existing ones, otherwise, use discovered subnets
+		var existingSubnetIDs []string
+		var subnetCidrs []string
+		for _, az := range v2LoadBalancer.AvailabilityZones {
+			existingSubnetIDs = append(existingSubnetIDs, *az.SubnetId)
+		}
+		if len(existingSubnetIDs) > 0 {
```

Comment:
under which case will this condition be true?
shouldn't lbs always have subnets?

Reply:
I was thinking about if it's a new NLB. but checking through the ensureLoadBalancerv2(), if it's a new NLB it will use the discoveredSubnetIDs. So yeah there should always be subnets. I'll update the PR